platooning_strategic_IHP/src/platoon_strategic_ihp.cpp

```diff
@@ -3363,12 +3343,39 @@ namespace platoon_strategic_ihp
                         ROS_DEBUG_STREAM("The target cutin pose is not on a valid lanelet. So no lane change!");
                         break;
                     } 
-                    int target_lanelet_id = target_lanelets[0].second.id(); //12301
+                    int target_lanelet_id = target_lanelets[0].second.id();
                     ROS_DEBUG_STREAM("target_lanelet_id: " << target_lanelet_id);
 
-                    // note: Since lanelet ID is not important for arbitrary lanechange, just use first lanelet's Id to create a maneuver msg.
-                    resp.new_plan.maneuvers.push_back(composeLaneChangeManeuverMessage(current_downtrack_, lc_end_dist,  
+                    lanelet::ConstLanelet starting_lanelet = wm_->getMap()->laneletLayer.get(current_lanelet_id);
+                    lanelet::ConstLanelet ending_lanelet = wm_->getMap()->laneletLayer.get(target_lanelet_id);
+
+                    auto relation = wm_->getMapRoutingGraph()->routingRelation(starting_lanelet, ending_lanelet);
+                    bool lanechangePossible = false;
+                    
+                    if (relation == lanelet::routing::RelationType::Left || relation == lanelet::routing::RelationType::Right)
```

Comment:
I think this check could fail if the lane change was planned across a lanelet boundary. Fortunately, since this is a strategic plugin, it is also enforcing its own assumptions, so it probably won't break at the moment. However, I think you should add a comment documenting this assumption and an enhancement issue on github to resolve it with a link to that issue in the code.